pkg/controller/vitesskeyspace/reconcile_resharding.go

```diff
@@ -13,25 +15,69 @@ import (
 	"vitess.io/vitess/go/vt/wrangler"
 )
 
+const (
+	maxSafeVReplicationLag = 10
+)
+
 func (r *ReconcileVitessKeyspace) reconcileResharding(ctx context.Context, vtk *planetscalev2.VitessKeyspace) error {
 	wr, ts, err := newWrangler(ctx, vtk.Spec.GlobalLockserver)
 	if err != nil {
 		return err
 	}
 	defer ts.Close()
 
-	workflows, err := wr.ListAllWorkflows(ctx, vtk.Spec.Name)
+	workflowList, err := wr.ListAllWorkflows(ctx, vtk.Spec.Name)
 	if err != nil {
-		workflows = make([]string, 0)
+		workflowList = make([]string, 0)
 	}
 
-	vtk.Status.ActiveWorkflows = workflows
-	// TODO: Address when ConditionUnknown is applicable.
-	if len(workflows) != 0 {
-		vtk.Status.ReshardingInProgress = corev1.ConditionTrue
-	} else {
-		vtk.Status.ReshardingInProgress = corev1.ConditionFalse
+	reshardingInProgress := corev1.ConditionUnknown
+	workflows := make([]planetscalev2.WorkflowStatus, 0, len(workflowList))
+	for _, workflowName := range workflowList {
+		workflow, err := wr.ShowWorkflow(ctx, workflowName, vtk.Spec.Name)
+		if err != nil {
+			return err
+		}
+		workflowStatus := planetscalev2.WorkflowStatus{
+			Workflow: workflow.Workflow,
+		}
+
+		for name, status := range workflow.ShardStatuses {
+			if status.MasterIsServing {
+				shard := strings.Split(name, "/")[0]
```

Comment:
What's the format of the key in the ShardStatuses map? Is that format guaranteed to stay the same forever? Is there a function in Vitess to parse it for us so we don't have to duplicate the logic in the operator?

Reply:
The format is `shard/tablet-alias`. I don't know if it will stay the same forever. There is not a function in Vitess to parse it for us.

Reply:
Can there ever be more than one entry in the map for the same shard, but different tablet aliases? If there can only be one per shard, is it too late to change the format upstream to be just the shard name and put the tablet alias inside the struct? Or are we the first ones to try to use this struct?

Reply:
No, there can't be, because we are only iterating over master tablets and we only have 1 master per shard. I would be happy to change the format upstream but I think this did get included in 7.0 and wasn't even identified as experimental so we might need to think about backwards compatibility. I'm not sure if anyone is depending on this yet. As of that release, this only effected what got logged though, so I can't imagine there's a strong dependency from a user.

Reply:
Does the code know how to ignore false masters?

Reply:
It just gets masters from topo and trusts that as source of truth.